src/cargo/core/resolver/mod.rs

```diff
@@ -369,21 +403,32 @@ impl<T> RcVecIter<T> {
     }
```

Comment:
If we're refactoring anyway, the only thing this is used for is to get the equivalent of `first()` on the underlying vec. Maybe just use that in the first place?